src/sessions/ses-langlocalice.py

```diff
@@ -0,0 +1,148 @@
+even_INT_FILELIST = []
+I_LIST_dir = root + mainLanguage + '/intact/'
+tmp = [str(s) for s in Path(I_LIST_dir).rglob('*wav')]
+INT_FILELIST = sorted(tmp, key=lambda x: int(x.split("/")[-1].split("_")[0]))
+for i in np.arange(len(INT_FILELIST)):
+    if int(INT_FILELIST[i].split("/")[-1].split("_")[0]) % 2 == 0:
+        even_INT_FILELIST.append(INT_FILELIST[i])
+    else:
+        odd_INT_FILELIST.append(INT_FILELIST[i])
+
+# get all DEGRADED
+odd_D_FILELIST = []
+even_D_FILELIST = []
+D_LIST_dir  = root + mainLanguage + '/degraded/'
+tmp = [str(s) for s in Path(D_LIST_dir).rglob('*wav')]
+D_FILELIST = sorted(tmp, key=lambda x: int(x.split("/")[-1].split("_")[0]))
+for i in np.arange(len(D_FILELIST)):
+    if int(D_FILELIST[i].split("/")[-1].split("_")[0]) % 2 == 0:
+        even_D_FILELIST.append(D_FILELIST[i])
+    else:
+        odd_D_FILELIST.append(D_FILELIST[i])
+
+# get all FOREIGN
+odd_F_FILELIST = []
+even_F_FILELIST = []
+F_LIST_dir  = root + foreignLangauge + '/intact/'
+tmp = [str(s) for s in Path(F_LIST_dir).rglob('*wav')]
+F_FILELIST = sorted(tmp, key=lambda x: int(x.split("/")[-1].split("_")[0]))
+for i in np.arange(len(F_FILELIST)):
+    if int(F_FILELIST[i].split("/")[-1].split("_")[0]) % 2 == 0:
+        even_F_FILELIST.append(F_FILELIST[i])
+    else:
+        odd_F_FILELIST.append(F_FILELIST[i])
+
+
+def get_audios_alice(C_LIST):
+    
+    order = ["X","I",'D','F','F','D',"I","X", "I",'D','F','F','D',"I","X",
+                 'X', 'D', 'F', 'I', 'I', 'F','D', 'X', 'D', 'F', 'I', 'I', 'F', 'D', 'X',
+                 'X', 'F','I', 'D', 'D', 'I', 'F','X', 'F', 'I', 'D','D', 'I', 'F','X']
+    
+    STIM_LIST = []
+    c_o_i = 0 
+    c_o_d = 0 
+    c_e_i = 0 
+    c_e_d = 0 
+    c_o_f = 0 
+    c_e_f = 0 
+    
+    for trial, cond in enumerate(order):
+
+        if cond == 'X':
+            #STIM_LIST.append('X') ===== TMP to debu - should be 12s of silence!
+            STIM_LIST.append('/Users/valentinaborghesani/Documents/4_Projects/CNeuromod/All_Tasks/task_stimuli/data/language/localizer/alice/English/intact/1_intact.wav')
+
+        if cond == 'I':
+            if C_LIST == 1:
+                STIM_LIST.append(odd_INT_FILELIST[c_o_i])
+                c_o_i = c_o_i+1
+            else:
+                STIM_LIST.append(even_INT_FILELIST[c_e_i])
+                c_e_i = c_e_i+1
+
+        if cond == 'D':
+            if C_LIST == 1:
+                STIM_LIST.append(even_D_FILELIST[c_e_d])
+                c_e_d = c_e_d+1
+            else:
+                STIM_LIST.append(odd_D_FILELIST[c_o_d]) 
+                c_o_d = c_o_d+1
+
+        if cond == 'F':
+            if C_LIST == 1:
+                STIM_LIST.append(even_F_FILELIST[c_e_f])
+                c_e_f = c_e_f+1
+            else:
+                STIM_LIST.append(odd_F_FILELIST[c_o_f])  
+                c_o_f = c_o_f+1
+                
+    return STIM_LIST
+
+STIM_LIST = get_audios_alice(C_LIST)
+
+TASKS = []
+for run_N in range(1, 3):
```

Comment:
I'm not sure if that's you meant to implement such design.
Each item TASK is supposed to be a separate fMRI run, each waiting for a start trigger.
For now it seems that each trial is made into a separate fMRI run.